inc/sitemaps/class-sitemaps.php

```diff
@@ -623,4 +614,34 @@ public static function get_post_statuses( $type = self::SITEMAP_INDEX_TYPE ) {
 
 		return $post_statuses;
 	}
+
+	/**
+	 * Sends all the required HTTP Headers
+	 *
+	 * @param string $type Provide a type for a post_type sitemap, SITEMAP_INDEX_TYPE for the root sitemap.
+	 */
+	private function send_headers( $type ) {
+		$headers = array(
+			$this->http_protocol . ' 200 OK'                                                       => 200,
```

Comment:
I think `http_response_code` will do the same. In that case you won't need the if check in the foreach below

Reply:
This allows filtering it to a 404 though, which could be useful...

Reply:
(Or a 301 redirect)